test/intl402/DateTimeFormat/constructor-options-dateStyle-valid.js

```diff
@@ -0,0 +1,34 @@
+// Copyright 2019 Igalia, S.L. All rights reserved.
+// This code is governed by the BSD license found in the LICENSE file.
+
+/*---
+esid: sec-initializedatetimeformat
+description: >
+    Checks handling of the options argument to the DateTimeFormat constructor.
+info: |
+    InitializeDateTimeFormat ( dateTimeFormat, locales, options )
+
+    ...
+    28. Let dateStyle be ? GetOption(options, "dateStyle", "string", « "full", "long", "medium", "short" », undefined).
+    29. If dateStyle is not undefined, set dateTimeFormat.[[DateStyle]] to dateStyle.
+---*/
+
+
+const validOptions = [
+  [undefined, undefined],
+  ["full", "full"],
+  ["long", "long"],
+  ["medium", "medium"],
+  ["short", "short"],
+  [{ toString() { return "full"; } }, "full"],
+  [{ valueOf() { return "long"; }, toString: undefined }, "long"],
+];
+for (const [dateStyle, expected] of validOptions) {
+  const dtf = new Intl.DateTimeFormat("en", { dateStyle });
+  const options = dtf.resolvedOptions();
+  assert.sameValue(options.dateStyle, expected);
+  assert.sameValue(
+    Object.getOwnPropertyDescriptor(options, "dateStyle") === undefined,
+    expected === undefined,
```

Comment:
this is a very odd assertion. I wonder if there is any way to improve this? I'll take a look again tomorrow.